build_tools/setup_helpers/extension.py

```diff
@@ -82,12 +81,15 @@ def build_extension(self, ext):
         ]
 
         # Default to Ninja
-        if 'CMAKE_GENERATOR' not in os.environ:
+        if 'CMAKE_GENERATOR' not in build_env or platform.system() == 'Windows':
             cmake_args += ["-GNinja"]
+        if platform.system() == 'Windows':
+            build_env.setdefault('CC', 'cl')
+            build_env.setdefault('CXX', 'cl')
```

Comment:
I think, instead of manipulating the environment variables here we can simply pass the following arguments to `cmake` command.

```
-DCMAKE_C_COMPILER=cl
-DCMAKE_CXX_COMPILER=cl
```

Reply:
we can test this once we get the windows build right